include/xtensor/xfiles_array.hpp

```diff
@@ -0,0 +1,233 @@
+#ifndef XTENSOR_FILES_ARRAY_HPP
+#define XTENSOR_FILES_ARRAY_HPP
+
+#include <istream>
+#include <fstream>
+#include <iostream>
+#include <vector>
+#include <array>
+
+#include "xarray.hpp"
+#include "xcsv.hpp"
+#include "xio.hpp"
+
+namespace xt
+{
+    template <class S>
+    void reopen(S& stream, const std::string& path)
+    {
+        if (stream.is_open())
+        {
+            stream.close();
+            stream.clear();
+        }
+        stream.open(path);
+    }
+
+    template <class T1, class T2, class S1, class S2>
+    void remap(T1** file_array, T2& array, S1& in_stream, S2& out_stream)
```

Comment:
Pointers and pointers to pointers should be avoided if not in very low-level code. What prevents from using references?

Besides this function seems to be called in access methods, having a lot of memory deletion and allocation will be a performance issue.

Reply:
Yes, I can use a reference.
You're right, the current implementation is really not production-ready, it is more like an attempt to have something working and to discuss choices. I will explain some of them.